Assignment4/javierdcp-071980560/Task07.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# **Task 07: Querying RDF(s)**
+
+# In[46]:
+
+
+get_ipython().system('pip install rdflib')
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+
+# First let's read the RDF file
+
+# In[47]:
+
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# In[48]:
+
+
+# TO DO
+ns = Namespace("http://somewhere#")
+#
+print("SPARQL")
+#for r in g.query(q1):
+from rdflib.plugins.sparql import prepareQuery
+
+q1 = prepareQuery('''
+  SELECT  ?sub  
+  WHERE {
+    ?sub rdfs:subClassOf ns:LivingThing.
+  }  
+  ''', initNs={"ns": ns}
+                  )
+
+for r in g.query(q1):
+    print(r.sub)
+
+#
+print("RDFLib")
+ns = Namespace("http://somewhere#")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+  print(s)
+
+
+# **TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+# 
+
+# In[49]:
+
+
+# TO DO
+
+print("SPARQL")
+
+q2 = prepareQuery('''
+  SELECT  ?instance  
+  WHERE {
+    {?sub rdfs:subClassOf* ns:Person.
+    ?instance a ?sub}
+  }  
+  ''', initNs={"ns": ns}
+                  )
+# Visualize the results
+for r in g.query(q2):
+    print(r.instance)
+
+print("with RDFLib")
+for s, p, o in g.triples((None, RDF.type, ns.Person)):
+    print(s)
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+    for s1, p1, o1 in g.triples((None, RDF.type, s)):
+        print(s1)
+
+
+# **TASK 7.3: List all individuals of "Person" or "Animal" and all their properties including their class with RDFLib and SPARQL. You do not need to list the individuals of the subclasses of person**
+# 
+
+# In[50]:
+
+
+# TO DO
+
+print("SPARQL")
+q3 = prepareQuery('''
+    SELECT DISTINCT ?individual ?property ?value 
+    WHERE {
+        ?individual rdf:type ns:Person.
+        
+        ?individual ?property ?value.
+        
+    }
+    ''', initNs={"ns": ns}
```

Comment:
this query should be merged with the one below, there is no need of having two queries for solving this exercise